### PR TITLE
Updates to elcord to make it a minor mode, add Windows support, et al.

### PR DESCRIPTION
**Pull request by Zulu-Inuoe**
An overview of all changes:

1. elcord is now a minor mode, rather than connecting automatically on startup
2. elcord now works on Windows, by use of a PowerShell script to serve as proxy between Emacs and Discord
3. Exported customization options via defcustom: - asdasd
   * Client ID if the user wants to use their own Discord Application
   * Extensible mappings for major modes to strings
   * Extensible mappings for major modes to icons
   * Hide buffer name and line if the user wishes to do so.
4. elcord now displays Emacs as the main icon on Discord, with a small icon for the currently active Major Mode
5. elcord now only updates Rich Presence every 15 seconds as requested by the [spec](https://discordapp.com/developers/docs/rich-presence/how-to#updating-presence), rather than every keystroke.
6. elcord gracefully handles starting up without Discord running, or if the Discord application exits, by attempting periodically to reconnect.
7. I've updated the README accordingly, adding documentation on the features and installation instructions.

### CODE REVIEW

**Comment by hithroc** [comment]
Can you add Haskell to the language list?

**Comment by Zulu-Inuoe** [comment]
@hithroc : see https://github.com/Zulu-Inuoe/elcord/commit/09ddf080fbe9e25183bdfb269464c8edca6e37ae
Even though I missed mentioning it, I added Haskell support there

I won't do another PR until this one gets merged, though

File: elcord.el [hunk 1]
```diff
@@ -1,154 +1,391 @@
...
-;;; Copyright 2017 heatingdevice
+;; Copyright (C) 2017 heatingdevice
 
-;;; Version: 0.0.2
-;;; Author: heatingdevice
-;;; URL: https://github.com/mstrodl/elcord
-;;; License: MIT
+;; Author: heatingdevice
+;;      Wilfredo Velázquez-Rodríguez <zulu.inuoe@gmail.com>
+;; Maintainer: Wilfredo Velázquez-Rodríguez <zulu.inuoe@gmail.com>
```
Comment (Mstrodl) [inline elcord.el]: Uhh

File: elcord.el [hunk 2]
```diff
@@ -1,154 +1,391 @@
...
+;; Created: 21 Nov 2017
+;; Version: 1.0.0
+;; Keywords: games
+;; Homepage: https://github.com/Zulu-Inuoe/elcord
```
Comment (Mstrodl) [inline elcord.el]: ????

**Comment by purcell** [comment]
Cool. Is this going to get merged, then? I'd love to close https://github.com/melpa/melpa/pull/5132 by pointing the MELPA recipe at this repo.

**Pull request merged**